scripts/mbedtls_framework/test_driver.py

```diff
@@ -0,0 +1,219 @@
+        src_include_dir_name = entries[0].name
+
+        if (self.dst_dir / "include").exists():
+            shutil.rmtree(self.dst_dir / "include")
+
+        if (self.dst_dir / "src").exists():
+            shutil.rmtree(self.dst_dir / "src")
+
+        if exclude_files is None:
+            exclude_files = set()
+
+        for file in iter_code_files(src_dir):
+            if any(fnmatch(file.name, pattern) for pattern in exclude_files):
+                continue
+            dst = self.dst_dir / file.relative_to(src_dir)
+            dst.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copy2(file, dst)
+
+        self.test_driver_include_dir = self.dst_dir / "include" / self.driver
+        (self.dst_dir / "include" / src_include_dir_name).rename( \
+             self.test_driver_include_dir)
+
+        headers = {
+            f.relative_to(self.test_driver_include_dir).as_posix() \
+            for f in self.test_driver_include_dir.rglob("*.h")
+        }
+        for f in iter_code_files(self.dst_dir):
+            self.__rewrite_inclusions_in_file(f, headers, \
+                                              src_include_dir_name, self.driver)
+
+    def get_identifiers_with_prefixes(self, prefixes: Set[str]):
+        """
+        Return the identifiers in the test driver that start with any of the given
+        prefixes.
+
+        All exposed identifiers are expected to start with one of these prefixes.
+        The returned set is therefore a superset of the exposed identifiers that
+        need to be prefixed.
+        """
+        identifiers = set()
+        for file in iter_code_files(self.dst_dir):
+            identifiers.update(run_ctags(file))
+
+        identifiers_with_prefixes = set()
+        for identifier in identifiers:
+            if any(identifier.startswith(prefix) for prefix in prefixes):
+                identifiers_with_prefixes.add(identifier)
+        return identifiers_with_prefixes
+
+    def prefix_identifiers(self, identifiers_to_prefix: Set[str]):
+        """
+        In all test driver files, prefix each identifier in `identifiers_to_prefix`
+        with the test driver prefix: <DRIVER>_ for uppercase identifiers,
+        and <driver>_ for lowercase ones.
+        """
+        for f in iter_code_files(self.dst_dir):
+            self.__prefix_identifiers_in_file(f, identifiers_to_prefix, self.driver)
+
+    @staticmethod
+    def __rewrite_inclusions_in_file(file: Path, headers: Set[str],
+                                     src_include_dir: str, driver: str,) -> None:
+        """
+        Rewrite `#include` directives in `file` that refer to `src_include_dir/...`
+        so that they instead refer to `driver/...`.
+
+        For example:
+            #include "mbedtls/private/aes.h"
+        becomes:
+            #include "libtestdriver1/private/aes.h"
+        """
+        include_line_re = re.compile(
+            fr'^\s*#\s*include\s*([<"])\s*{src_include_dir}/([^>"]+)\s*([>"])',
```

Comment:
You can't have spaces inside the quotes/brackets.

```suggestion
            fr'^\s*#\s*include\s*([<"]){src_include_dir}/([^>"]+)([>"])',
```